### ISSUE TITLE
test-with-comment flags any test when any comment exists anywhere in the document

### ISSUE THREAD

**Issue by VasilevNStas** [objectionary/lints#1375]
## Problem

`test-with-comment` flags **any** top-level test whenever **any** comment exists anywhere in the document — the relation between a comment and the test it documents is not checked at all.

## Root cause

`src/main/resources/org/eolang/lints/tests/test-with-comment.xsl:14`:

```xml
<xsl:for-each select="/object/o[eo:test-name(@name) and /object/comments/comment]">
```

The second conjunct is a document-global existential check: if the document contains a comment (e.g. documenting an ordinary object `app`), every top-level test gets a defect "must have a comment".

## Minimal example

```eo
+package x
+architect yegor@256.com
+version 0.0.1
+home https://github.com/objectionary
+spdx SPDX-FileCopyrightText: Copyright (c) 2026 Objectionary.com
+spdx SPDX-License-Identifier: MIT

# This is app.
[] > app
  x > @

[] > +can-add-test
  42 > @
```

`+can-add-test` is reported as missing a comment, even though the comment clearly documents `app`, not the test.

## Expected behavior

A comment should be attributed to the object it precedes/belongs to (by position/line or by adjacency), and the lint should flag only tests that have no comment *of their own*.

## Related

- `src/test/resources/org/eolang/lints/packs/single/test-with-comment/*.yaml` — existing packs only cover the happy paths.

**Comment by 0crat** [comment]
@vasilevnstas Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+710](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
